packages/front-end/pages/saved-groups/[sgid].tsx

```diff
@@ -105,26 +110,110 @@ export default function EditSavedGroupPage() {
     [mutate, savedGroup],
   );
 
+  const savedGroupsReferencingTarget = useMemo(() => {
+    if (!savedGroup || !allSavedGroups) return savedGroup ? [savedGroup] : [];
+
+    const targetId = savedGroup.id;
+    const referencingGroups: SavedGroupInterface[] = [];
+
+    function extractSavedGroupIds(obj: unknown): Set<string> {
+      const ids = new Set<string>();
+
+      if (typeof obj !== "object" || obj === null) return ids;
+
+      if (Array.isArray(obj)) {
+        obj.forEach((item) => {
+          extractSavedGroupIds(item).forEach((id) => ids.add(id));
+        });
+        return ids;
+      }
+
+      for (const [key, value] of Object.entries(obj)) {
+        if (key === "$savedGroups") {
+          if (Array.isArray(value)) {
+            value.forEach((id) => {
+              if (typeof id === "string") ids.add(id);
+            });
+          } else if (typeof value === "string") {
+            ids.add(value);
+          }
+        } else if (typeof value === "object" && value !== null) {
+          extractSavedGroupIds(value).forEach((id) => ids.add(id));
+        }
+      }
+
+      return ids;
+    }
+
+    allSavedGroups.forEach((sg) => {
+      if (sg.id === targetId) return;
+
+      const condition = sg.condition;
+      if (!condition || condition === "{}") return;
+
+      try {
+        const parsed = JSON.parse(condition);
+        const savedGroupIds = extractSavedGroupIds(parsed);
+        if (savedGroupIds.has(targetId)) {
+          referencingGroups.push(sg);
+        }
+      } catch (e) {
+        console.error("Error parsing condition", e);
+      }
+    });
```

Comment:
I'm ok with this logic, but we could simplify and just look to see if the condition string contains this quoted saved group id. Tiny chance of false positives, but much simpler logic to read and maintain.